tensorboard/plugins/core/core_plugin.py

```diff
@@ -135,6 +142,16 @@ def _send_404_without_logging(self, request):
   def _redirect_to_index(self, unused_request):
     return utils.redirect('/')
 
+  @wrappers.Request.application
+  def _serve_if_experiment(self, apps, path_experiment, path_no_experiment, request):
+    """Conditionally serve paths based on whether there is an experiment id."""
+    has_experiment = plugin_util.experiment_id(request.environ)
```

Comment:
Let's call this `experiment_id_from_path` or something that indicates it's an experiment ID rather than a boolean, even if we're only using it in a boolean context right now.